File: finished/Plus One.js
// https://leetcode.com/problems/plus-one/

/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  let index = digits.length - 1;

  while (index >= 0) {
    digits[index]++;

    if (digits[index] < 10) return digits;

    digits[index] = 0;
    index--;
  }

  digits.unshift(1);

  return digits;
};

File: src/python/finished/plus_one.py
from typing import List

# https://leetcode.com/problems/plus-one/


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1

        for i in reversed(range(len(digits))):
            digits[i] += carry
            if digits[i] < 10:
                carry = 0
                break

            digits[i] = 0
            carry = 1

        if carry:
            digits.insert(0, 1)

        return digits
